src/api/cat.js

```diff
@@ -1,7 +1,32 @@
 'use strict'
 
-const argCommand = require('../cmd-helpers').argCommand
+const bs58 = require('bs58')
+const isIPFS = require('is-ipfs')
+const promisify = require('promisify-es6')
 
 module.exports = (send) => {
-  return argCommand(send, 'cat')
+  const cat = promisify((multihash, callback) => {
+    try {
+      multihash = cleanMultihash(multihash)
+    } catch (err) {
+      return callback(err)
+    }
+    send('cat', multihash, null, null, function (err, result) {
+      if (err) {
+        return callback(err)
+      }
+      return callback(null, result)
+    })
+  })
+  return cat
+}
+
+function cleanMultihash (multihash) {
+  if (!isIPFS.multihash(multihash)) {
+    throw new Error('not valid multihash')
+  }
+  if (Buffer.isBuffer(multihash)) {
```

Comment:
>The linked to doc says "the raw Buffer of the multihash (or of and encoded version)". Did I misunderstand what "of the encoded version" means? It means "base58 encoded", right?

I don't think the raw Buffer is base58 encoded, this is borrowed from the object.get definition of a multihash input. The string mentioned after it is the base58 encoded multihash

>Underyling assumption: this cat method is supposed to satisfy interface-ipfs-core's cat method. Is this correct?

Right, this upgrade to cat will now take a buffer input so that it is interop with interface-core